inst/shiny-server/server/upload.R

```diff
@@ -6,65 +6,96 @@ DataList <- reactiveValues(data = DataSetList())
 # set up list of datasets (scan the repository, looking for .rds files)
 observe({
   repo_dir <<- get_repo_location()
-  rds_files <- list.files(repo_dir, pattern = '.rds') %>% sub('\\.rds$', '', .)
-
+  if (input$respository.type == 'PBO') {
+    rds_files <- list.files(repo_dir, pattern = '.rds') %>% sub('\\.rds$', '', .)
```

Comment:
I would remove `%>%`... up to you. I really don't like `magrittr` now

Reply:
In cases like this I prefer to still use it for readability, we just need to prevent using it within loops and basic functions which are executed often